### PR TITLE
[READY]Pulling claw 2

### PR DESCRIPTION
**Pull request by arsserpentarium**
:cl: Selea
add: pulling claw.Integrated circuit, which allow assemblies to pull things.
/:cl:
Pulling claw is integrated circuit,which allows drones to pull things.
As i promised, pulling claw will go in separate PR

### CODE REVIEW

File: code/modules/integrated_electronics/subtypes/manipulation.dm [hunk 1]
```diff
@@ -349,6 +349,44 @@
 	set_pin_data(IC_OUTPUT, 3, contents.len)
 	push_data()
 
+/obj/item/integrated_circuit/manipulation/claw
+	name = "pulling claw"
+	desc = "A circuit which can pull things.."
```
Comment (praisenarsie) [inline code/modules/integrated_electronics/subtypes/manipulation.dm]: remove a .

Comment (arsserpentarium) [reply code/modules/integrated_electronics/subtypes/manipulation.dm]: AAAAAAAAaaaaaAAAAAAaaaaAAAaAaAaaAaaaaaaaaa!!!!!!!111!!1111

**Comment by kevinz000** [comment]
honestly just refactor pulling to work from any atom movable

**Comment by arsserpentarium** [comment]
It brokes light this way.Also I am not sure how to generalize pulling to work from AM, because as you can see, pulling in obj is slightly different from mobs.

File: code/game/machinery/magnet.dm [hunk 2]
```diff
@@ -21,7 +21,7 @@
 	var/code = 0 // frequency code, they should be different unless you have a group of magnets working together or something
 	var/turf/center // the center of magnetic attraction
 	var/on = FALSE
-	var/pulling = 0
+	var/pulling1 = 0
```
Comment (nicbn) [inline code/game/machinery/magnet.dm]: Rename to something more readable, like magnetpulling.

Comment (nicbn) [reply code/game/machinery/magnet.dm]: Also 0 and 1 to FALSE/TRUE.

**Review by nicbn — CHANGES_REQUESTED** [review]

File: code/game/machinery/magnet.dm (hunk 2 above)
Comment (arsserpentarium) [reply code/game/machinery/magnet.dm]: mkay.I'll fix it, when there will be several more reviews to less clutter commit history.

**Comment by nicbn** [comment]
Why did you redefine all pull procs for /obj? Just move it out of /mob to /atom/moveable or to be a global proc which is called on both /obj and /mob pull procs.

**Comment by arsserpentarium** [comment]
because:
1.pulling proc for mob and obj slightly different.slightly, but incompatible.So if I'll put it into atom/moveable, there still will be these pieces of code.But one in AM, one in mob,instead of one in obj, oe in mob.
2.it broken light somehow, when I tried last time.

**Comment by arsserpentarium** [comment]
anything else?

**Comment by Bawhoppen** [comment]
Please explain what your PR is in the description???

**Comment by calzilla1** [comment]
Automatic spacing bot when

File: code/game/objects/objs.dm [hunk 3]
```diff
@@ -63,6 +66,85 @@
 		visible_message("<span class='danger'>[src] shatters into a million pieces!</span>")
 		qdel(src)
 
+/obj/proc/start_pulling(atom/movable/AM,gs)
```
Comment (ninjanomnom) [inline code/game/objects/objs.dm]: @nicbn is completely right here, it should be moved to `/atom/movable`

Just saying it doesn't work isn't an excuse. You're a coder, make it work.

File: code/game/machinery/magnet.dm (hunk 2 above)
Comment (ninjanomnom) [reply code/game/machinery/magnet.dm]: I commented on this before in the other pr, why not keep the name you changed it to there?

**Review by ninjanomnom — DISMISSED** [review]

**Comment by arsserpentarium** [comment]
1. @calzilla1 autospacing with drones will be possible.But not very soon. Because drone can't navigate on station and can't open doors. But I think, it will be possible after next PR.Not sure.
2.2 ninja.
a.already fixed,I'll push it soon.
b.I'll try, but it might take time. I have some other problems now.

**Comment by arsserpentarium** [comment]
uhh...looks like I messed.
You can't resist assembly's reinforced grip.

**Comment by arsserpentarium** [comment]
one more commit is pending. have problems with git right now

**Comment by The-clown-known-as-Pepe** [comment]
oh no drones will be able to slip people

**Comment by arsserpentarium** [comment]
Hm.really. I haven't thought about autoslip drone. Which is putting and picking banana peel every step. To pull someone towards peel. Fabolous.
Brilliant idea!

**Comment by epochayur** [comment]
my dream of making a corpse-finder bot is slowly becoming a reality

**Comment by ShizCalev** [comment]
Conflict.

**Comment by arsserpentarium** [comment]
conflict was fixed

**Comment by arsserpentarium** [comment]
It compiles.Looks like,travis is glitching

File: code/game/atoms_movable.dm [hunk 4]
```diff
@@ -27,6 +27,8 @@
 	var/datum/forced_movement/force_moving = null	//handled soley by forced_movement.dm
 	var/floating = FALSE
 	var/movement_type = GROUND		//Incase you have multiple types, you automatically use the most useful one. IE: Skating on ice, flippers on water, flying over chasm/space, etc.
+	var/atom/movable/pulling = null
```
Comment (Cyberboss) [inline code/game/atoms_movable.dm]: remove the = null

File: code/game/atoms_movable.dm [hunk 5]
```diff
@@ -61,7 +63,91 @@
 			return FALSE
 	return ..()
 
+/atom/movable/proc/start_pulling(atom/movable/AM,gs)
+	if(!AM || !src)
+		return FALSE
```
Comment (Cyberboss) [inline code/game/atoms_movable.dm]: Change this to just !QDELETED(AM)

**Review by Cyberboss — DISMISSED** [review]

File: code/game/atoms_movable.dm (hunk 5 above)
Comment (arsserpentarium) [reply code/game/atoms_movable.dm]: are you sure, that !QDELETED(AM), not QDELETED(AM) ?

**Comment by arsserpentarium** [comment]
So,will this PR be merged ever?

**Comment by arsserpentarium** [comment]
@Cyberboss

**Comment by optimumtact** [comment]
you need to fix your comment entry for the changelog

**Comment by arsserpentarium** [comment]
Is it right now?

**Pull request merged**